### ISSUE TITLE
Add Cypress utility methods for OS / Platform / Arch / Browser / Browser Version

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress#824]
Something like this...

```js
// utility functions
Cypress.isPlatform()
Cypress.isBrowser()
Cypress.isArch()
Cypress.isBrowserVersion()

// constants
Cypress.arch
Cypress.platform
Cypress.browser
Cypress.browserVersion

if (Cypress.browser === "chrome") {

}

if (Cypress.isPlatform(["win32", "darwin"]) {

})

if (!Cypress.isBrowser("firefox") {

}
```

What about normalizing things like `chrome`, `chromium`, `canary` ?

**Comment by bahmutov** [comment]
I really think we should split this into `platform + arch` vs `browser` and for now just do the first part - this is needed to correctly test `cy.exec` for example.

When adding any functions must update `cypress/cli/index.d.ts` too

**Comment by brian-mann** [comment]
Fixed by https://github.com/cypress-io/cypress/pull/986

**Comment by brian-mann** [comment]
Released in [`1.1.3`](https://docs.cypress.io/guides/references/changelog.html#1-1-3).
